orchestra/bots/staffbot.py

```diff
@@ -26,28 +32,75 @@ class StaffBot(BaseBot):
         (r'staff (?P<task_id>[0-9]+)', 'staff'),
         (r'restaff (?P<task_id>[0-9]+) (?P<username>[\w.@+-]+)', 'restaff'),
     )
+    task_does_not_exist_error = 'Task {} does not exist'
+    task_assignment_error = 'Task {} got an error: "{}"'
+    worker_does_not_exist = 'Worker with username {} does not exist'
+    task_assignment_does_not_exist_error = (
+        'TaskAssignment associated with user {} and task {} does not exist.')
 
     def __init__(self, **kwargs):
         default_config = getattr(settings, 'STAFFBOT_CONFIG', {})
         default_config.update(kwargs)
         token = settings.SLACK_STAFFBOT_TOKEN
         super().__init__(token, **kwargs)
 
-    def staff(self, task_id):
+    def staff(self, task_id, **kwargs):
         """
         This function handles staffing a request for the given task_id.
         """
+        try:
+            task = Task.objects.get(id=task_id)
+            required_role = role_required_for_new_task(task)
+            request_cause = kwargs.get('request_cause',
+                                       StaffingRequest.RequestCause.USER.value)
+        except Task.DoesNotExist:
+            logger.exception(self.task_does_not_exist_error.format(task_id))
+            return self.format_slack_message(
+                self.task_does_not_exist_error.format(task_id))
+        except TaskAssignmentError as error:
+            logger.exception(self.task_assignment_error.format(task_id, error))
+            return self.format_slack_message(
+                self.task_assignment_error.format(task_id, error))
+
+        self._send_task_to_workers(task, required_role, request_cause)
         return self.format_slack_message('Staffed task {}!'.format(task_id))
 
-    def restaff(self, task_id, username):
+    def restaff(self, task_id, username, **kwargs):
         """
         This function handles restaffing a request for the given task_id.
         The current user for the given username is removed, and a new user
         is found.
         """
-        return self.format_slack_message(
-            'Restaffed task {} for {}!'.format(task_id, username)
-        )
+        # TODO(kkamalov): maybe username could also be slack_username
+        try:
+            task = remove_worker_from_task(username, task_id)
+            required_role = role_required_for_new_task(task)
+            request_cause = kwargs.get('request_cause',
+                                       StaffingRequest.RequestCause.USER.value)
+        except Worker.DoesNotExist:
```

Comment:
same for error messages here, i think it's cleaner to assign a variable and pass that to the slack message/logger